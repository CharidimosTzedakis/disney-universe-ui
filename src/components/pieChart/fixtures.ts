export const pieChartDataPoints = [
  {
    name: "Mikey",
    y: 2,
    films: ["Terminator", "Back to the future"],
  },
  {
    name: "Donald",
    y: 4,
    films: ["Blade runner", "film2", "film3", "film4"],
  },
  {
    name: "Taz",
    y: 7,
    films: ["film5", "film6", "film7", "film8", "film9", "film10", "film11"],
  },
  { name: "char1", y: 0, films: [] },
];
