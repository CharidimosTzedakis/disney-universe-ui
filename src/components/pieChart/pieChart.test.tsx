import { expect, describe, it } from "vitest";
import { render } from "@testing-library/react";
import CharactersPieChart from "@components/pieChart";
import { pieChartDataPoints } from "@test/fixtures";
import useChartsStore from "@stores/chartStore";

describe("<CharactersPieChart/>", () => {
  it("should render correctly", () => {
    useChartsStore.getState().filmsPieChart.data = [...pieChartDataPoints];
    const { container } = render(<CharactersPieChart />);

    expect(container).toMatchSnapshot();
  });
});
