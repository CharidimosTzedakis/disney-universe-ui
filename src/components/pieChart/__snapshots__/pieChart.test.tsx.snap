// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CharactersPieChart/> > should render correctly 1`] = `
<div>
  <div
    class="_pieChartContainer_00ea6c"
  >
    <div
      data-highcharts-chart="0"
      style="overflow: hidden;"
    >
      <div
        class="highcharts-container "
        dir="ltr"
        id="highcharts-6kjljnl-0"
        style="position: relative; overflow: hidden; width: 600px; height: 400px; text-align: left; line-height: normal; z-index: 0; user-select: none; outline: none; padding: 0px;"
      >
        <svg
          aria-label="Number of films for characters"
          class="highcharts-root"
          height="400"
          role="img"
          style="font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", sans-serif; font-size: 1rem;"
          version="1.1"
          viewBox="0 0 600 400"
          width="600"
          xmlns="http://www.w3.org/2000/svg"
        >
          <desc>
            Created with Highcharts 12.0.2
          </desc>
          <defs>
            <filter
              id="highcharts-drop-shadow-0"
            >
              <fedropshadow
                dx="1"
                dy="1"
                flood-color="#000000"
                flood-opacity="0.75"
                stdDeviation="2.5"
              />
            </filter>
          </defs>
          <rect
            class="highcharts-background"
            fill="#ffffff"
            filter="none"
            height="400"
            rx="0"
            ry="0"
            width="600"
            x="0"
            y="0"
          />
          <rect
            class="highcharts-plot-background"
            fill="none"
            filter="none"
            height="375"
            width="580"
            x="10"
            y="10"
          />
          <rect
            class="highcharts-plot-border"
            data-z-index="1"
            fill="none"
            height="375"
            stroke="#cccccc"
            stroke-width="0"
            width="580"
            x="10"
            y="10"
          />
          <g
            class="highcharts-series-group"
            data-z-index="3"
            filter="none"
          />
          <text
            class="highcharts-title"
            data-z-index="4"
            style="font-size: 1.2em; color: rgb(51, 51, 51); font-weight: bold; fill: #333333;"
            text-anchor="middle"
            transform="translate(0,0) scale(1 1)"
            transform-origin="10 NaN"
            x="300"
            y="10"
          >
            Number of films for characters
          </text>
          <text
            class="highcharts-subtitle"
            data-z-index="4"
            style="color: rgb(102, 102, 102); font-size: 0.8em; fill: #666666;"
            text-anchor="middle"
            transform="translate(0,0) scale(1 1)"
            transform-origin="10 NaN"
            x="300"
            y="10"
          />
          <text
            class="highcharts-caption"
            data-z-index="4"
            style="color: rgb(102, 102, 102); font-size: 0.8em; fill: #666666;"
            text-anchor="start"
            transform="translate(0,0) scale(1 1)"
            transform-origin="10 NaN"
            x="10"
            y="385"
          />
          <g
            class="highcharts-legend highcharts-no-tooltip"
            data-z-index="7"
            visibility="hidden"
          >
            <rect
              class="highcharts-legend-box"
              fill="none"
              filter="none"
              height="8"
              rx="0"
              ry="0"
              stroke="#999999"
              stroke-width="0"
              width="8"
              x="0"
              y="0"
            />
            <g
              data-z-index="1"
            >
              <g />
            </g>
          </g>
          <text
            class="highcharts-credits"
            data-z-index="8"
            style="cursor: pointer; color: rgb(153, 153, 153); font-size: 0.6em; fill: #999999;"
            text-anchor="end"
            x="590"
            y="395"
          >
            Highcharts.com
          </text>
        </svg>
      </div>
    </div>
    <button
      class="ant-btn css-dev-only-do-not-override-142vneq ant-btn-primary ant-btn-color-primary ant-btn-variant-solid _exportToExcelBtn_00ea6c"
      type="button"
    >
      <span>
        Export to excel
      </span>
    </button>
  </div>
</div>
`;
